app/controllers/departments_controller.rb

```diff
@@ -0,0 +1,57 @@
+class DepartmentsController < ApplicationController
+
+
+  def index
+    @departments = Department.sorted
+    render (:index)
+  end
+
+  def show
+    @department = Department.find(params[:id])
+  end
+
+  def new
+    @department = Department.new
+  end
+
+  def create
+    @department = Department.new(department_params)
+    if @department.save
+      flash[:notice] = "Department created successfully!"
```

Comment:
We will use i18n for such notice or sentence (`"Department created successfully!"`). Please study on i18n.